CMakeLists.txt

```diff
@@ -33,6 +27,11 @@ nanobind_add_module(
   ${ISL_SOURCES}
 )
 
+# Work around https://github.com/inducer/islpy/issues/120.
+# See https://stackoverflow.com/questions/43554227/extern-inline-func-results-in-undefined-reference-error
+# for some context.
+set_source_files_properties(${ISL_SOURCES} PROPERTIES COMPILE_DEFINITIONS __OPTIMIZE_SIZE__)
```

Comment:
Out of curiosity: Why is it that the compiler isn't setting that define?

Reply:
IIUC, it's an "old glibc" type of situation?

Reply:
Huh, I just realized that we're specifically asking the compiler to *not* optimize size. So glibc was just broken when you do that?

Reply:
Btw, sorry about the stream-of-consciousness comments here, I'm just working my way towards an understanding of the issue.

Reply:
> Out of curiosity: Why is it that the compiler isn't setting that define?

This is only set automatically by the compiler when compiling with `-Os`, but not the other optimization options, which is why `-Os` (set by default, when not specifying `NOMINSIZE`) worked out of the box.

> IIUC, it's an "old glibc" type of situation?

Yes (IIUC).

> Huh, I just realized that we're specifically asking the compiler to not optimize size. So glibc was just broken when you do that?

Not really, defining `__OPTIMIZE_SIZE__` (as in this PR) only disables certain inline implementations (incl. the `strtoimax` implementations mentioned in #120) for header files in older glibc versions (`inttypes.h` for `strtoimax`). I guess there may be a bug in how the compiler handles those `extern inline` implementations, or they may have been missing a `static`. I'm not sure. 

See also https://github.com/bminor/glibc/blob/9e4e896f0f5a19a16c1a77567463b013a0f4952d/include/features.h#L515-L520 for a bit more context.

Reply:
Hmm, OK. Thanks for explaining, at least to the extent that we understand. Since it seems to do the job, I'm mostly happy,  I think. :slightly_smiling_face: